From 1e2c145d77c482d90fb92ead8ad8c42d14beadb5 Mon Sep 17 00:00:00 2001
From: Jenkins Slave <jenkins@code.rdkcentral.com>
Date: Wed, 11 Oct 2023 16:20:14 +0100
Subject: [PATCH] Remove hal dependencies

---
 source/Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/source/Makefile.am b/source/Makefile.am
index acee0c5..5c1eada 100644
--- a/source/Makefile.am
+++ b/source/Makefile.am
@@ -32,5 +32,5 @@ rdkledmanager_DEPENDENCIES = \
 
 rdkledmanager_CFLAGS = -Werror -D_ANSC_LINUX -D_ANSC_USER -D_ANSC_LITTLE_ENDIAN_ -DFEATURE_SUPPORT_RDKLOG $(DBUS_CFLAGS) $(SYSTEMD_CFLAGS)
 rdkledmanager_SOURCES = led_manager.c led_manager_events.c led_manager_utils.c led_manager_dbus_utils.c
-rdkledmanager_LDFLAGS = -lccsp_common -lrdkloggers -ljson-c $(DBUS_LIBS) $(SYSTEMD_LDFLAGS) -lsysevent -lhal_platform -lledmanager -lsky_hub
+rdkledmanager_LDFLAGS = -lccsp_common -lrdkloggers -ljson-c $(DBUS_LIBS) $(SYSTEMD_LDFLAGS) -lsysevent -lhal_platform
 rdkledmanager_LDADD =  $(rdkledmanager_DEPENDENCIES)
-- 
2.37.3

