diff --git a/source/LedManager/Makefile.am b/source/LedManager/Makefile.am
index fbff526..2a9e693 100644
--- a/source/LedManager/Makefile.am
+++ b/source/LedManager/Makefile.am
@@ -40,7 +40,7 @@
 
 rdkledmanager_CFLAGS = -Werror -D_ANSC_LINUX -D_ANSC_USER -D_ANSC_LITTLE_ENDIAN_ -DFEATURE_SUPPORT_RDKLOG $(DBUS_CFLAGS) $(SYSTEMD_CFLAGS)
 rdkledmanager_SOURCES = led_manager.c led_manager_events.c led_manager_utils.c led_manager_dbus_utils.c
-rdkledmanager_LDFLAGS = -lccsp_common -lrdkloggers -ljson-c $(DBUS_LIBS) $(SYSTEMD_LDFLAGS) -lsysevent -lhal_platform -lledmanager -lsky_hub -lsyscfg
+rdkledmanager_LDFLAGS = -lccsp_common -lrdkloggers -ljson-c $(DBUS_LIBS) $(SYSTEMD_LDFLAGS) -lsysevent -lhal_platform -lsyscfg
 if FEATURE_LEDMGR_WEBCONFIG
 rdkledmanager_SOURCES += ledmgr_webconfig.c
 rdkledmanager_LDFLAGS += -lpthread -lrbus -lsyscfg -lmsgpackc -lwebconfig_framework -lcjson
